### PR TITLE
Feature/nar 1975 support eddsa

### PR DESCRIPTION
**Pull request by Ptroger**
- Adds support for EDDSA
  - generation
  - sign
  - verify
  - signJwt
  - hex/JWK conversions

This requires bumping SDK version

### CODE REVIEW

File: packages/signature/src/lib/__test__/unit/sign.spec.ts [hunk 1]
```diff
@@ -110,6 +121,23 @@ describe('sign', () => {
     expect(signature).toBe('afu_-8eYXRpHAt_nTVRksRmiwVZpq7iC2rBVhGQT5YcJlViKV9wD3OIlRYAxa7JkNd1Yqzf_x2ohLzqGjmlb2hs')
   })
 
+  it('should sign ED25519 correctly', async () => {
```
Comment (wcalderipe) [inline packages/signature/src/lib/__test__/unit/sign.spec.ts:124]:
```suggestion
  it('signs ED25519 correctly', async () => {
```

File: packages/signature/src/lib/__test__/unit/util.spec.ts [hunk 2]
```diff
@@ -58,6 +65,15 @@ const k1Jwk: Jwk = {
   d: 'ENTv9xjMG6RkHwamtOk3l0mlHOZN7herKFEmGPPKK04'
 }
 
+const eddsaKey: Jwk = {
```
Comment (wcalderipe) [inline packages/signature/src/lib/__test__/unit/util.spec.ts:68]:
```suggestion
const eddsaPrivateKey: Jwk = {
```

Make it clear it's a private key, please.

File: packages/signature/src/lib/__test__/unit/util.spec.ts [hunk 3]
```diff
@@ -170,6 +208,23 @@ describe('publicKeyToJwk', () => {
...
     expect(jwk2.kid).toBe('myKey2')
   })
+
+  it('converts a valid EDDSA hex public key to JWK', async () => {
+    const jwk = publicKeyToJwk(eddsaHexPublicKey, Alg.EDDSA)
+    const { d: _d, ...eddsaPublicKey } = eddsaKey
+    expect(jwk).toEqual(eddsaPublicKey)
+  })
+
+  it('works', async () => {
```
Comment (wcalderipe) [inline packages/signature/src/lib/__test__/unit/util.spec.ts:218]: Best test description that I've read in a long time 🤣  "well.. it. just works!"

```suggestion
  it('ensures hex consistency across standard and polyfill implementations', async () => {
```

File: packages/signature/src/lib/__test__/unit/verify.spec.ts [hunk 4]
```diff
@@ -203,6 +205,23 @@ describe('verifySecp256k1', () => {
...
+    const key = ed.utils.randomPrivateKey()
+    const pubKey = await ed.getPublicKey(key)
+
+    const jwk = publicKeyToJwk(toHex(pubKey), Alg.EDDSA)
+    const signature = await ed.sign(msg, key)
+
+    const isVerified = await ed.verify(signature, msg, pubKey)
+    const isVerifiedByUs = await verifyEd25519(signature, msg, jwk)
+
+    expect(isVerified).toEqual(true)
```
Comment (wcalderipe) [inline packages/signature/src/lib/__test__/unit/verify.spec.ts:220]: What are you testing here?

File: packages/signature/src/lib/utils.ts [hunk 5]
```diff
@@ -103,6 +123,33 @@ export const secp256k1PrivateKeyToJwk = (privateKey: Hex, keyId?: string): Secp2
   }
 }
 
+export const ed25519PublicKeyToJwk = (publicKey: Hex, keyId?: string): Ed25519PublicKey => {
+  // Remove 0x prefix if present
+  const hexPubKey = publicKey.slice(2)
```
Comment (wcalderipe) [inline packages/signature/src/lib/utils.ts:128]: Should we write a unit test for it? If that invariant is important, we should consider capturing it in a test case.

**Review by wcalderipe — APPROVED** [review]

File: packages/signature/src/lib/__test__/unit/util.spec.ts (hunk 3 above)
Comment (Ptroger) [reply packages/signature/src/lib/__test__/unit/util.spec.ts:218]: Lol, forgot this one

File: packages/signature/src/lib/__test__/unit/util.spec.ts (hunk 2 above)
Comment (Ptroger) [reply packages/signature/src/lib/__test__/unit/util.spec.ts:68]: It's both, it holds private and public material (it has 'd' AND 'x')

File: packages/signature/src/lib/utils.ts (hunk 5 above)
Comment (Ptroger) [reply packages/signature/src/lib/utils.ts:128]: This is an invariant from library usage, noble expects hex strings stripped from their prefix, internally, we use hex definition from viem which requires the prefix.

**Pull request merged**